cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+            if isinstance(self.step_size, (int, float)):
+                if self.step_size < 0:
+                    raise ValueError(
+                        'step_size must be > 0, found {}'.format(self.step_size)
+                    )
+            else:
+                if len(self.step_size) != chains:
+                    raise ValueError(
+                        'number of step_sizes must match number of chains '
+                        ' found {} step_sizes for {} chains '.format(
+                            len(self.step_size), chains
+                        )
+                    )
+                for i in range(len(self.step_size)):
+                    if self.step_size[i] < 0:
+                        raise ValueError(
+                            'step_size must be > 0, found {}'.format(
+                                self.step_size[i]
+                            )
+                        )
+
+        if self.metric is not None:
+            dims = None
+            if isinstance(self.metric, str):
+                if self.metric in ['diag', 'diag_e']:
+                    self.metric = 'diag_e'
+                elif self.metric in ['dense', 'dense_e']:
+                    self.metric = 'dense_e'
+                else:
+                    if not os.path.exists(self.metric):
+                        raise ValueError('no such file {}'.format(self.metric))
+                    dims = read_metric(self.metric)
+            elif isinstance(self.metric, list):
+                if len(self.metric) != chains:
+                    raise ValueError(
+                        'number of metric files must match number of chains '
+                        ' found {} metric files for {} chains '.format(
+                            len(self.metric), chains
+                        )
+                    )
+                names_set = set(self.metric)
+                if len(names_set) != len(self.metric):
+                    raise ValueError(
+                        'each chain must have its own metric file,'
+                        ' found duplicates in metric files list.'
+                    )
+                for i in range(len(self.metric)):
+                    if not os.path.exists(self.metric[i]):
+                        raise ValueError(
+                            'no such file {}'.format(self.metric[i])
+                        )
+                    if i == 0:
+                        dims = read_metric(self.metric[i])
+                    else:
+                        dims2 = read_metric(self.metric[i])
+                        if len(dims) != len(dims2):
+                            raise ValueError(
+                                'metrics files {}, {},'
+                                ' inconsistent metrics'.format(
+                                    self.metric[0], self.metric[i]
+                                )
+                            )
+                        for j in range(len(dims)):
```

Comment:
for j, dim in enumerate(dims):

are these one 1D?

Reply:
dims will either be length 1 (diag metric) or length 2 (dense metric)